shuup/gdpr/anonymizer.py

```diff
@@ -0,0 +1,162 @@
+        "name_ext": "blank_value",
+        "merchant_notes": "blank_value",
+        "street": "random_string",
+        "street1": "random_string",
+        "street2": "random_string",
+        "street3": "random_string",
+        "street_address": "random_string",
+        "tax_number": "random_number",
+        "city": "random_string",
+        "longitude": "none_value",
+        "latitude": "none_value",
+        "birth_date": "none_value",
+        "data": "none_value",
+        "email": "null_email",
+        "username": "random_username"
+    }
+
+    def random_string(self, len=8):
+        return get_random_string(len)
+
+    def none_value(self):
+        return None
+
+    def blank_value(self):
+        return ""
+
+    def null_zip(self):
+        return get_random_string(6)
+
+    def null_ip(self):
+        return "0.0.0.0"
+
+    def null_email(self):
+        return "%s@%s.com" % (get_random_string(10), get_random_string(8))
+
+    def null_name(self):
+        return get_random_string(8)
+
+    def null_phone(self):
+        return get_random_string(9, "1234567890")
+
+    def random_number(self):
+        return int(get_random_string(9, "123456789"))
+
+    def random_username(self):
+        return get_random_string(20, "1234567890")
+
+    def anonymize_contact(self, pk):
+        contact = Contact.objects.get(pk=pk)
+
+        # make sure to remove from marketing before we lost the contact email
+        contact.marketing_permission = False
+        contact.save()
+
+        # anonymize all saved addresses
+        for saved_address in contact.savedaddress_set.all():
+            self.anonymize_object(saved_address.address)
+
+        # anonymize all orders
+        for order in contact.customer_orders.all():
+            self.anonymize_order(order)
+
+        # anonymize all Front baskets
+        for basket in contact.customer_baskets.all():
+            self.anonymize_object(basket)
+
+        # anonymize all Core baskets
+        for basket in contact.customer_core_baskets.all():
+            self.anonymize_object(basket)
+
+        self.anonymize_object(contact.default_shipping_address)
+        self.anonymize_object(contact.default_billing_address)
+        self.anonymize_object(contact)
+
+        contact.is_active = False
+        contact.save()
+
+    def anonymize_company(self, company):
+        if not company:
+            return
+        assert isinstance(company, CompanyContact)
+        self.anonymize_contact(company.id)
+
+    def anonymize_person(self, person):
+        if not person:
+            return
+        assert isinstance(person, PersonContact)
+
+        for order in person.orderer_orders.all():
+            self.anonymize_order(order)
+
+        for basket in person.orderer_baskets.all():
+            self.anonymize_object(basket)
+
+        for basket in person.orderer_core_baskets.all():
+            self.anonymize_object(basket)
+
+        self.anonymize_contact(person.pk)
```

Comment:
Should we somewhere here also anonymize the user?

Reply:
I let the programmer decide where and how to anonymize the user, it is called in the view when anonymizing the customer

Reply:
@Pikkupomo is this really correct? If the merchant now anonymize contact he needs to separately know whether there is user linked to the contact and anonymize that too? Isn't user linked to person contact anonymized every single time when the contact is anonymized?

Reply:
Check here @tulimaki  https://github.com/shuup/shuup/pull/1253/files#diff-4dba3d2446f86fab35b2b5abd2e57102R122